nautobot/extras/models/groups.py

```diff
@@ -0,0 +1,313 @@
+"""Dynamic Groups Models."""
+
+import logging
+
+from django import forms
+from django.contrib.contenttypes.models import ContentType
+from django.core.exceptions import ValidationError
+from django.core.serializers.json import DjangoJSONEncoder
+from django.db import models
+from django.urls import reverse
+
+from nautobot.core.models.generics import OrganizationalModel
+from nautobot.extras.groups import dynamicgroup_map_factory
+from nautobot.extras.utils import extras_features
+from nautobot.utilities.querysets import RestrictedQuerySet
+
+
+logger = logging.getLogger(__name__)
+
+
+class DynamicGroupQuerySet(RestrictedQuerySet):
```

Comment:
Would this be better placed in `nautobot/extras/querysets.py`?

Reply:
Yes.